sky/backends/backend_utils.py

```diff
@@ -779,19 +778,12 @@ def write_cluster_config(
     # for the validation.
     # TODO(tian): Move more cloud agnostic vars to resources.py.
     resources_vars = to_provision.make_deploy_variables(cluster_name_on_cloud,
-                                                        region, zones)
+                                                        region, zones, dryrun)
     config_dict = {}
 
-    azure_subscription_id = None
-    if isinstance(cloud, clouds.Azure):
-        azure_subscription_id = cloud.get_project_id(dryrun=dryrun)
-
-    gcp_project_id = None
-    if isinstance(cloud, clouds.GCP):
-        gcp_project_id = cloud.get_project_id(dryrun=dryrun)
-
     specific_reservations = set(
-        skypilot_config.get_nested(('gcp', 'specific_reservations'), set()))
+        skypilot_config.get_nested(
+            (str(to_provision.cloud).lower(), 'specific_reservations'), set()))
```

Comment:
Should we update other places for the path, e.g. https://github.com/skypilot-org/skypilot/blob/6efc906f69dcb546542e824cb9549f185425a960/sky/optimizer.py#L253

Reply:
Good catch! Updated. Also, simplified the logic of reservations, as we don't have to pass the `specific_reservations` all around.